api/client/application/client_test.go

```diff
@@ -330,107 +353,131 @@ func (s *applicationSuite) TestDestroyApplications(c *gc.C) {
 		},
 	}}
 	delay := 1 * time.Minute
-	client := newClient(func(objType string, version int, id, request string, a, response interface{}) error {
-		c.Assert(request, gc.Equals, "DestroyApplication")
-		c.Assert(a, jc.DeepEquals, params.DestroyApplicationsParams{
-			Applications: []params.DestroyApplicationParams{
-				{ApplicationTag: "application-foo", Force: true, MaxWait: &delay},
-				{ApplicationTag: "application-bar", Force: true, MaxWait: &delay},
-			},
-		})
-		c.Assert(response, gc.FitsTypeOf, &params.DestroyApplicationResults{})
-		out := response.(*params.DestroyApplicationResults)
-		*out = params.DestroyApplicationResults{expectedResults}
-		return nil
-	})
-	results, err := client.DestroyApplications(application.DestroyApplicationsParams{
+	result := new(params.DestroyApplicationResults)
+	results := params.DestroyApplicationResults{expectedResults}
+	args := params.DestroyApplicationsParams{
+		Applications: []params.DestroyApplicationParams{
+			{ApplicationTag: "application-foo", Force: true, MaxWait: &delay},
+			{ApplicationTag: "application-bar", Force: true, MaxWait: &delay},
+		},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DestroyApplication", args, result).SetArg(2, results).Return(nil)
+
+	client := application.NewClientFromCaller(mockFacadeCaller)
+	res, err := client.DestroyApplications(application.DestroyApplicationsParams{
 		Applications: []string{"foo", "bar"},
 		Force:        true,
 		MaxWait:      &delay,
 	})
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(results, jc.DeepEquals, expectedResults)
+	c.Assert(res, jc.DeepEquals, expectedResults)
 }
 
 func (s *applicationSuite) TestDestroyApplicationsArity(c *gc.C) {
-	client := newClient(func(objType string, version int, id, request string, a, response interface{}) error {
-		return nil
-	})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	result := new(params.DestroyApplicationResults)
+	results := params.DestroyApplicationResults{}
+	args := params.DestroyApplicationsParams{
+		Applications: []params.DestroyApplicationParams{
+			{ApplicationTag: "application-foo"},
+		},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DestroyApplication", args, result).SetArg(2, results).Return(errors.New("expected 1 result(s), got 0"))
```

Comment:
The "expected 1 results(s), not 0" does result from the mock result, it should come from api code when the number of request does not equal the number of results received, see line 530.